src/main/java/seedu/address/model/person/Person.java

```diff
@@ -94,13 +104,14 @@ public boolean equals(Object other) {
                 && phone.equals(otherPerson.phone)
                 && email.equals(otherPerson.email)
                 && address.equals(otherPerson.address)
-                && tags.equals(otherPerson.tags);
+                && tags.equals(otherPerson.tags)
+                && Objects.equals(appointment, otherPerson.appointment); // Include appointment in equality check
```

Comment:
could appointment.equals(otherPerson.appointment) work? think narrowing the scope might be safer

Reply:
Agreed, resolved